### PR TITLE
update readme and add traffic pattern sql

### PR DESCRIPTION
**Pull request by chmnata**
## What this pull request accomplishes:

- 

## Issue(s) this solves:

- #937 

## What, in particular, needs to reviewed:

- Traffic pattern sql, does the mean of 5 min bin make sense? Or should I change it to one hour?

### CODE REVIEW

File: here/traffic_patterns/README.md [hunk 1]
```diff
@@ -128,5 +128,9 @@ LATERAL (
     ) )
 );
 ```
-## Sad news
-For top secret contract reasons, we are no longer receiving traffic patterns data. As of writing this (in August 2022) the latest (and last) traffic patterns dataset that we received is from 2019. As time goes by, and the street network evolves, traffic patterns will cease to be useful...
\ No newline at end of file
+## Generate Quarterly traffic patterns
+For top secret contract reasons, we are no longer receiving traffic patterns data. The latest (and last) traffic patterns dataset that we received is from 2019. As time goes by, and the street network evolves, traffic patterns will cease to be useful. We wil be creating quaterly traffic patterns for each version for routing purposes. 
+
+Depending on the date you are routing, you should use different quaters, years, and map version of traffic patterns. We will only include weekday (Tue - Thurs), and exclude holiday seasons to create a more [generalized traffic pattern layer](/sql/create_quaterly_traffic_patterns.sql).
```
Comment (radumas) [inline here/traffic_patterns/README.md:134]: quaters -> quarters

File: here/traffic_patterns/sql/create_quaterly_traffic_patterns.sql [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        dt >= _start_date AND dt < _end_date
+        AND EXTRACT(ISODOW FROM dt) IN (2, 3, 4) -- only include tues-thurs traffic
+        AND holiday.dt IS NULL -- excluding holidays
+    GROUP BY routing_streets.link_dir, dt, hr
+)
+
+SELECT
+    link_dir,
+    hr,
+    harmean(daily_cost)::int AS cost
```
Comment (radumas) [inline here/traffic_patterns/sql/create_quaterly_traffic_patterns.sql:22]: this isn't a cost it's a speed, should be renamed to speed

**Review by radumas — CHANGES_REQUESTED** [review]
This look pretty good. One pretty important change in terms of explicitly naming columns in the traffic patterns table.

Otherwise I would also suggest specifying what need to be changed in the two queries to make traffic pattern for a particular quarter and revision. Should also make explicit the assumptions in terms of if no data then speed limit and if no speed limit then 50 km/hr

**Comment by chmnata** [comment]
Ready for re-review @radumas, made them into functions instead. :meow_dio:

File: here/traffic_patterns/sql/create_routing_speed_limit.sql [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    EXECUTE format($$CREATE TABLE here.%I AS
+                    SELECT
+                        routing_streets.link_dir,
+                        COALESCE(
+                            CASE
+                                WHEN RIGHT(routing_streets.link_dir, 1) = 'T' AND att.to_spd_lim > 0 THEN att.to_spd_lim
+                                WHEN RIGHT(routing_streets.link_dir, 1) = 'F' AND att.fr_spd_lim > 0 THEN att.fr_spd_lim
+                                ELSE NULL
+                            END, 50
```
Comment (radumas) [inline here/traffic_patterns/sql/create_routing_speed_limit.sql:26]: `COALESCE(CASE [...]  ELSE NULL END, 50)`, the last part feels redundant

**Review by radumas — CHANGES_REQUESTED** [review]
Smol fix requested

File: here/traffic_patterns/sql/create_routing_speed_limit.sql (hunk 3 above)
Comment (chmnata) [reply here/traffic_patterns/sql/create_routing_speed_limit.sql]: got rid of COALESCE

**Review by radumas — APPROVED** [review]
:party-porg:

**Pull request merged**